core/src/rules/deindent_package_directives.rs

```diff
@@ -0,0 +1,168 @@
+use crate::prelude::*;
+
+pub struct DeindentPackageDirectives {}
+
+impl LogicalLinesConsolidator for DeindentPackageDirectives {
+    fn consolidate(&self, (tokens, lines): (&mut [Token], &mut [LogicalLine])) {
+        match tokens
+            .iter()
+            .map(TokenData::get_token_type)
+            .find(|tt| !tt.is_comment_or_directive())
+        {
+            Some(TokenType::Keyword(KeywordKind::Package)) => {}
+            _ => return,
+        }
+
+        for line in lines {
+            if let Some(TokenType::CompilerDirective | TokenType::ConditionalDirective(_)) = line
+                .get_tokens()
+                .first()
+                .and_then(|t_idx| tokens.get(*t_idx))
+                .map(TokenData::get_token_type)
+            {
```

Comment:
Is there a reason you don't do something like this?

```suggestion
            if let LogicalLineType::CompilerDirective | LogicalLineType::ConditionalDirective = line.get_line_type() {
```

Reply:
I didn't think of it